File: app/src/main/java/com/example/githubuserssample/activities/UserDetailsActivity.java
package com.example.githubuserssample.activities;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;
import com.example.githubuserssample.R;
import com.example.githubuserssample.models.User;
import com.example.githubuserssample.models.UserDetails;
import com.example.githubuserssample.networking.RetrofitClient;
import com.google.android.material.appbar.MaterialToolbar;
import com.mikhaellopez.circularimageview.CircularImageView;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class UserDetailsActivity extends AppCompatActivity {
    TextView fullName, bio, repos, followers, following;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_details);

        MaterialToolbar materialToolbar = findViewById(R.id.toolbar);
        setSupportActionBar(materialToolbar);
        materialToolbar.setNavigationOnClickListener(v -> onBackPressed());

        fullName = findViewById(R.id.user_full_name);
        bio = findViewById(R.id.user_bio);
        repos = findViewById(R.id.user_repos_count);
        followers = findViewById(R.id.user_followers_count);
        following = findViewById(R.id.user_following_count);

        User user = (User) getIntent().getSerializableExtra("user");
        Log.e("USER", user.toString());
        CircularImageView userAvi = findViewById(R.id.user_avi);

        Glide.with(this)
                .load(user.getAvatarUrl())
                .into(userAvi);

        getUserDetails(user.getLogin());


    }

    private void getUserDetails(String login) {
        RetrofitClient.getInstance().getApi()
                .getUserDetails(login)
                .enqueue(new Callback<UserDetails>() {
                    @SuppressLint("SetTextI18n")
                    @Override
                    public void onResponse(Call<UserDetails> call, Response<UserDetails> response) {
                        if (response.code()==200){
                            UserDetails userDetails = response.body();

                            fullName.setText(userDetails.getName());
                            bio.setText(userDetails.getBio());
                            repos.setText(userDetails.getPublicRepos().toString());
                            followers.setText(userDetails.getFollowers().toString());
                            following.setText(userDetails.getFollowing().toString());
                        }
                    }

                    @Override
                    public void onFailure(Call<UserDetails> call, Throwable t) {

                    }
                });
    }


}

File: app/src/main/java/com/example/githubuserssample/activities/UserDetailsActivity.kt
package com.example.githubuserssample.activities

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.example.githubuserssample.R
import com.example.githubuserssample.models.User
import com.example.githubuserssample.models.UserDetails
import com.example.githubuserssample.networking.RetrofitClient
import kotlinx.android.synthetic.main.activity_user_details.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class UserDetailsActivity : AppCompatActivity() {
    private lateinit var user: User

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_details)

        toolbar.setNavigationOnClickListener { onBackPressed() }
        setSupportActionBar(toolbar)
        user = intent.getSerializableExtra("user") as User

        Log.e("USER", user.toString())

        Glide.with(this)
                .load(user.avatarUrl)
                .into(user_avi)

        userDetails

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_share, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_share) {

            val messageBody = "Check out this awesome developer @${user.login}. \n\n ${user.htmlUrl}."

            val intent = Intent().apply {
                action = Intent.ACTION_SEND
                putExtra(Intent.EXTRA_TEXT, messageBody)
                type = "text/plain"
            }

            startActivity(Intent.createChooser(intent, null))

        }

        return super.onOptionsItemSelected(item)

    }

    private val userDetails: Unit
        get() {
            RetrofitClient.apiService
                    .getUserDetails(user.login)
                    .enqueue(object : Callback<UserDetails> {
                        override fun onResponse(call: Call<UserDetails>, response: Response<UserDetails>) {
                            val userDetails = response.body()

                            user_bio.text = userDetails?.bio
                            user_full_name.text = userDetails?.name
                            user_repos_count.text = userDetails?.publicRepos.toString()
                            user_followers_count.text = userDetails?.followers.toString()
                            user_following_count.text = userDetails?.following.toString()

                        }

                        override fun onFailure(call: Call<UserDetails>, t: Throwable) {

                        }
                    })
        }

}